src/Decompiler/Scanning/IdentifierRelocator.cs

```diff
@@ -97,6 +97,8 @@ public Identifier VisitStackLocalStorage(StackLocalStorage loc)
 
         public Identifier VisitTemporaryStorage(TemporaryStorage tmp)
         {
+            if (tmp is GlobalStorage)
```

Comment:
This makes me think `GlobalStorage` merits its own visitor. You shouldn't have had to do this.